sw/device/lib/drivers/rv_plic/rv_plic.h

```diff
@@ -135,121 +158,171 @@ typedef enum dif_plic_irq_trigger {
    * Trigger on a level (when the signal remains high).
    */
   kDifPlicIrqTriggerLevel,
-} dif_plic_irq_trigger_t;
+} plic_irq_trigger_t;
+
 
 /**
- * Creates a new handle for PLIC.
- *
- * This function does not actuate the hardware.
- *
- * @param params Hardware instantiation parameters.
- * @param[out] plic Out param for the initialized handle.
- * @return The result of the operation.
- */
-DIF_WARN_UNUSED_RESULT
-dif_plic_result_t dif_plic_init(dif_plic_params_t params, dif_plic_t *plic);
+ * Enum for describing all the different types of interrupt
+ * sources that the rv_plic handles
+*/
+typedef enum irq_sources
+{
+  IRQ_UART_SRC,   // from 1 to 8
+  IRQ_GPIO_SRC,   // from 9 to 32 
+  IRQ_I2C_SRC,    // from 33 to 48
+  IRQ_SPI_SRC,    // line 49
+  IRQ_BAD = -1    // default failure case
+} irq_sources_t;
+
+
+/****************************************************************************/
+/**                                                                        **/
+/*                           EXPORTED VARIABLES                             */
+/**                                                                        **/
+/****************************************************************************/
 
 /**
- * Returns whether a particular interrupt is currently pending.
- *
- * @param plic A PLIC handle.
- * @param irq An interrupt type.
- * @param[out] is_pending Out-param for whether the interrupt is pending.
- * @return The result of the operation.
- */
-DIF_WARN_UNUSED_RESULT
-dif_plic_result_t dif_plic_irq_is_pending(const dif_plic_t *plic,
-                                          dif_plic_irq_id_t irq,
-                                          bool *is_pending);
+ * Flag used to handle the wait for interrupt.
+ * The core can test this variable to check if it has to wait
+ * for an interrupt coming from the RV_PLIC.
+ * When an interrupt occurs, this flag is set to 1 by the plic in order
+ * for the core to continue with the execution of the program.
+*/
+extern int8_t external_intr_flag;
```

Comment:
i do not understand why this is here: 1) it is under global section in C file, 2) this variable should be on the user-scope and not on the HAL

Reply:
I use it inside rv_plic.c so that the handler function can change it after calling the handler for the specific interrupt source. 
I also need it in the rv_plic.h so I can access it from the main (by including the header) and use it as a wile-loop variable when I call the wait for interrupt.

```
  while(external_intr_flag==0) {
      gpio_write(&gpio, GPIO_TB_OUT, true);
      wait_for_interrupt();
  }
```